website/blog/2018-04-04-using-typescript-with-react-native.md

```diff
@@ -0,0 +1,310 @@
+
+You will also need [Node.js](https://nodejs.org/en/), [NPM](https://www.npmjs.com), and [Yarn](https://yarnpkg.com/lang/en).
+
+## Initializing
+
+Once you've tried scaffolding out an ordinary React Native project, you'll be ready to start adding TypeScript. Let's go ahead and do that.
+
+```sh
+react-native init MyAwesomeProject
+cd MyAwesomeProject
+```
+
+## Adding TypeScript
+
+The next step is to add TypeScript to your project. The following commands will:
+
+* add TypeScript to your project
+* add [React Native TypeScript Transformer](https://github.com/ds300/react-native-typescript-transformer) to your project
+* initialize an empty TypeScript config file, which we'll configure next
+* add an empty React Native TypeScript Transformer config file, which we'll configure next
+* adds [typings](https://github.com/DefinitelyTyped/DefinitelyTyped) for React and React Native
+
+Okay let's go ahead and run these.
+
+```sh
+yarn add --dev typescript
+yarn add --dev react-native-typescript-transformer
+yarn tsc --init --pretty --jsx react
+touch rn-cli.config.js
+yarn add --dev @types/react @types/react-native
+```
+
+The `tsconfig.json` file contains all the settings for the TypeScript compile. The defaults created by the command above are mostly fine, but open the file and uncomment the following line:
+
+```js
+{
+  ...
+  // "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
+  ...
+}
+```
+
+You'll also want to add this to the **top-level** of the config, to tell TypeScript not to try compiling your `node_modules` directory:
+
+```js
+{
+  ...
+  "exclude": [
+    "node_modules"
+  ]
+  ...
+}
+```
+
+The `rn-cli.config.js` contains the settings for the React Native TypeScript Transformer. Open it and add the following:
+
+```js
+module.exports = {
+  getTransformModulePath() {
+    return require.resolve('react-native-typescript-transformer');
+  },
+  getSourceExts() {
+    return ['ts', 'tsx'];
+  },
+};
+```
+
+## Migrating to TypeScript
+
+Rename the generated `App.js` and `__tests_/App.js` files to `App.tsx`. `index.js` needs to use the `.js` extension. All new files should use the `.tsx` extension (or `.ts` if the file doesn't contain any JSX).
+
+If you tried to run the app now, you'd get an error like `object prototype may only be an object or null`. This is caused by a failure to import the default export from React as well as a named export on the same line. Open `App.tsx` and modify the import at the top of the file:
+
+```diff
```

Comment:
Whoa, I had not seen this in use before. Seems to work well in the GitHub preview, will have to see if Docusaurus's highlighter supports it well.

Reply:
It looks good locally as well as in the preview: 

<img width="864" alt="screen shot 2018-04-05 at 9 28 52 am" src="https://user-images.githubusercontent.com/498212/38368638-cb475b66-38b3-11e8-81aa-ae2d55e058d8.png">